core/src/main/scala/kafka/server/KafkaHealthcheck.scala

```diff
@@ -70,19 +72,40 @@ class KafkaHealthcheck(private val brokerId: Int,
   }
 
   /**
-   *  When we get a SessionExpired event, we lost all ephemeral nodes and zkclient has reestablished a
-   *  connection for us. We need to re-register this broker in the broker registry.
+   *  When we get a SessionExpired event, it means that we have lost all ephemeral nodes and ZKClient has re-established
+   *  a connection for us. We need to re-register this broker in the broker registry.
    */
-  class SessionExpireListener() extends IZkStateListener {
+  class SessionExpireListener extends IZkStateListener {
+
+    private[server] val stateToSensorMap = {
+      import KeeperState._
+      val stateToNameMap = Map(
+        Disconnected -> "disconnected",
+        SyncConnected -> "sync-connected",
+        AuthFailed -> "auth-failed",
+        ConnectedReadOnly -> "connected-read-only",
+        SaslAuthenticated -> "sasl-authenticated",
+        Expired -> "expired"
+      )
+      stateToNameMap.map { case (state, stateName) =>
+        val sensorName = s"zk-$stateName"
```

Comment:
Thanks for the feedback @fpj. I don't have a strong opinion either way and I'm happy to change it. Just to make sure we are all on the same page, @junrao, what do you think? Note that we have to decide on the name prefix as well as the tag group name a couple of lines below.